wurzel/datacontract/datacontract.py

```diff
@@ -130,7 +130,7 @@ def __hash__(self) -> int:
 
     def __eq__(self, other: object) -> bool:
         # pylint: disable-next=not-an-iterable
-        for field in self.model_fields:
+        for field in self.__class__.model_fields:
```

Comment:
Using type(self).model_fields instead of self.__class__.model_fields can improve readability and consistency.
```suggestion
        for field in type(self).model_fields:
```